app/src/main/java/nodomain/freeyourgadget/gadgetbridge/impl/GBDevice.java

```diff
@@ -281,15 +286,20 @@ public void setBatteryLevel(short batteryLevel) {
         }
     }
 
-    /**
-     * Returns a string representation of the battery state.
-     */
-    public String getBatteryState() {
-        return mBatteryState != null ? mBatteryState : GBApplication.getContext().getString(R.string._unknown_);
+    public BatteryState getBatteryState() {
+        return mBatteryState;
+    }
+
+    public void setBatteryState(BatteryState mBatteryState) {
+        this.mBatteryState = mBatteryState;
+    }
+
+    public short getBatteryThresholdPercent() {
+        return mBatteryThresholdPercent;
     }
 
-    public void setBatteryState(String batteryState) {
-        mBatteryState = batteryState;
+    public void setBatteryThresholdPercent(short batteryThresholdPercent) {
```

Comment:
I didn't meant it to be the device specific configuration, but rather the device-independent value. The idea is that each device could have its own custom threshold and call this method to trigger the GB device-independent behavior at the desired level.

ATM there is only the miband using this functionality, and for the miband the default threshold is good hence it doesn't change it (in fact at the moment this method is never called).